ServiceApp/src/org/droidplanner/core/mission/Mission.java

```diff
@@ -265,9 +269,13 @@ public void onMissionLoaded(List<msg_mission_item> msgs) {
     }
 
     private List<MissionItem> processMavLinkMessages(List<msg_mission_item> msgs) {
+        waypointToMissionItem.clear();
```

Comment:
@TShapinsky if a mission is pushed to the copter when it's unarmed, this method will be called when it's armed, causing the previous set of mission info to be cleared, and replaced. 

In most case that's fine, but complex mission items (survey, structure scan...) will be replaced with the set of simple mission items they are composed of, causing a loss of info about the complex mission items.